## ДЗ к Уроку 6. Взаимосвязь величин. Параметрические и непараметрические показатели корреляции. Корреляционный анализ.

### Задание 1.
Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [17]:
import numpy as np
import scipy.stats as stats
from statsmodels.stats.weightstats import _tconfint_generic as t_stat
from statsmodels.stats.weightstats import _zconfint_generic as z_stat

Найдём ковариацию двух величин по формулам:


In [5]:
zp =np.array ([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])



In [6]:
x_mean = zp.mean()
y_mean = ks.mean()
xy_mean = (zp*ks).mean()
print (x_mean, y_mean,xy_mean)

101.4 709.9 81141.7


In [7]:
cov_ks = xy_mean - x_mean * y_mean
cov_ks

9157.839999999997

Найдём ковариацию двух величин с помощью встроенных функций:

In [8]:
np.cov(zp,ks, ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [9]:
cov_ks2 = ((zp -zp.mean()) * (ks -ks.mean())).mean()
cov_ks2

9157.84

Найдём коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков

In [10]:
x_std = zp.std()
y_std = ks.std()


In [11]:
corr_ks = cov_ks / (x_std * y_std)
corr_ks

0.8874900920739158

Теперь найдём коэффициент корреляции Пирсона с помощью функции numpy

In [13]:
np.corrcoef(zp,ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

Ответ: 
ковариацию двух величин - 9157,84

коэффициент корреляции Пирсона - 0,88749009


## Задание 2

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [14]:
x =np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
x_mean = x.mean()
std_x = x.std(ddof=1)
x_mean_std = std_x / (np.sqrt (len(x)))

In [18]:
t_stat(x_mean, x_mean_std, len(x) - 1, 0.05, 'two-sided' )

(110.55608365158724, 125.64391634841274)

Ответ: 
Доверительный интервал располагается в диапозоне (110.556; 125.643)


## Задание 3

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [19]:
n = 27
x_mean = 174.2
x_std = np.sqrt(25)
x_mean_std = x_std / np.sqrt(n)


In [20]:
z_stat(x_mean, x_mean_std, 0.05, 'two-sided' )

(172.3140237765397, 176.08597622346028)

Ответ: Доверительный интервал располагается в диапозоне (172.314; 176.085)
